libvast/src/execution_node.cpp

```diff
@@ -316,38 +321,48 @@ struct exec_node_state : inbound_state_mixin<Input>,
     // Instantiate the operator with its input type.
     auto output_generator = op->instantiate(make_input_adapter(), *ctrl);
     if (not output_generator) {
-      return caf::make_error(
-        ec::unspecified, fmt::format("{} failed to instantiate operator: {}",
-                                     *self, output_generator.error()));
+      VAST_DEBUG("{} could not instantiate operator: {}", *self,
+                 output_generator.error());
+      return add_context(output_generator.error(),
+                         "{} failed to instantiate operator", *self);
     }
     if (not std::holds_alternative<generator<Output>>(*output_generator)) {
-      return caf::make_error(ec::logic_error,
-                             fmt::format("{} failed to instantiate operator",
-                                         *self));
+      return caf::make_error(
```

Comment:
Would `add_context(ec::logic_error, "...", ...)` work here as well?

Reply:
Ha, you are right. I mean, the intended use case for `add_context` was somewhat different, but it seems like it can also be used as a replacement for `caf::make_error`. I will not make this change right now for consistency reasons, but we could use that everywhere...